lib/utils/report.mjs

```diff
@@ -16,21 +16,28 @@ import { RULE_NAME_ALL } from '../constants.mjs';
  * @type {import('stylelint').Utils['report']}
  */
 export default function report(problem) {
-	const { node, index, endIndex, line, start, end, result, ruleName, word, ...rest } = problem;
+	const { node, index, endIndex, line, start, end, result, ruleName, word, fix, ...rest } = problem;
 	const {
 		disabledRanges,
 		quiet,
 		ruleSeverities,
 		config: { defaultSeverity = 'error', ignoreDisables } = {},
 		customMessages: { [ruleName]: message = rest.message } = {},
 		customUrls: { [ruleName]: customUrl } = {},
+		ruleMetadata: { [ruleName]: metadata } = {},
 	} = result.stylelint;
 	const { messageArgs = [], severity = ruleSeverities[ruleName] } = rest;
 	const ruleSeverity = isFn(severity) ? severity(...messageArgs) || defaultSeverity : severity;
 
 	// In quiet mode, mere warnings are ignored
 	if (quiet && ruleSeverity !== 'error') return;
 
+	if (isFn(fix) && metadata && !metadata.fixable) {
```

Comment:
question

If `metadata` is missing, this would never `throw`?

Reply:
Right. I assume plugins without metadata.

Reply:
However, it may be good to require metadata itself when using `fix()` 🤔

Reply:
But unfortunately, the following change fails the `checkAgainstRule()` test:

```diff
-	if (isFn(fix) && metadata && !metadata.fixable) {
+	if (isFn(fix) && !metadata?.fixable) {
```

```sh-session
$ npm -s run test-only -- lib/utils/__tests__/checkAgainstRule.test.mjs
...
  ● checkAgainstRule › handles array rule settings

    The "at-rule-empty-line-before" rule requires "meta.fixable" to be truthy if the "fix" callback is being passed
...
```

Because there are no metadata loaded in `checkAgainstRule()`:

https://github.com/stylelint/stylelint/blob/cb3a7f8d5e89f2e467a4166293c17d3c2d1e72cc/lib/utils/checkAgainstRule.mjs#L47